adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralNativeRenderer.m

```diff
@@ -0,0 +1,256 @@
+    
+    /// Holds the state for impression being logged.
+    atomic_flag _impressionLogged;
+    
+    /// The Mintegral native ad.
+    MTGBidNativeAdManager *_nativeManager;
+    
+    /// The Mintegral media view.
+    MTGMediaView *_mediaView;
+        
+    /// An ad event delegate to invoke when ad rendering events occur.
+    id<GADMediationNativeAdEventDelegate> _adEventDelegate;
+
+    /// The Mintegral ad unit ID.
+    NSString *_adUnitId;
+    
+    /// The Mintegral native ad data.
+    MTGCampaign *_campaign;
+    
+    /// Icon image.
+    GADNativeAdImage *_icon;
+    
+    /// Array of GADNativeAdImage objects.
+    NSArray<GADNativeAdImage *> *_images;
+}
+
+- (void)renderNativeAdForAdConfiguration:(nonnull GADMediationNativeAdConfiguration *)adConfiguration completionHandler:(nonnull GADMediationNativeLoadCompletionHandler)completionHandler {
+    // Store the ad config and completion handler for later use.
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationNativeLoadCompletionHandler originalCompletionHandler =
+        [completionHandler copy];
+    _adLoadCompletionHandler = ^id<GADMediationNativeAdEventDelegate>(
+        _Nullable id<GADMediationNativeAd> ad, NSError *_Nullable error) {
+      if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+        return nil;
+      }
+      id<GADMediationNativeAdEventDelegate> delegate = nil;
+      if (originalCompletionHandler) {
+        delegate = originalCompletionHandler(ad, error);
+      }
+      originalCompletionHandler = nil;
+      return delegate;
+    };
+    
+    NSString *adUnitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    UIViewController *rootViewController = adConfiguration.topViewController;
+    
+    if ([GADMAdapterMintegralUtils isStringEmpty:adUnitId]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID connot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+        
+    _adUnitId = adUnitId;
+    _nativeManager = [[MTGBidNativeAdManager alloc]initWithPlacementId:placementId unitID:adUnitId presentingViewController:rootViewController];
+    _nativeManager.delegate = self;
+    [_nativeManager loadWithBidToken:adConfiguration.bidResponse];
+}
+
+- (MTGMediaView *)createMediaView {
+    if (_mediaView) {
+        return _mediaView;
+    }
+    _mediaView= [[MTGMediaView alloc] initWithFrame:CGRectZero];
+    _mediaView.delegate = self;
+    return _mediaView;
+}
+
+#pragma mark - MTGBidNativeAdManagerDelegate
+- (void)nativeAdsLoaded:(nullable NSArray *)nativeAds bidNativeManager:(nonnull MTGBidNativeAdManager *)bidNativeManager {
+    
+    if ([nativeAds isKindOfClass:NSArray.class] && nativeAds.count > 0) {
+        _campaign = nativeAds.firstObject;
```

Comment:
Can you clarify the implementation here? Seems like `nativeAds` is an array of `MTGCampaign`, which is just the native ad data?

Reply:
yes,just native ad data